src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -187,7 +187,8 @@ public override BoundNode VisitITuplePattern(BoundITuplePattern node)
                         {
                             foreach (BoundPropertySubpattern subpattern in rp.Properties)
                             {
-                                if (subpattern.Member is BoundPropertySubpatternMember { HasErrors: false } member)
+                                if (subpattern.Member is BoundPropertySubpatternMember member
+                                    && member.Symbol.Kind is SymbolKind.Property or SymbolKind.Field)
```

Comment:
```cs
subpattern.Member is BoundPropertySubpatternMember { Symbol.Kind: SymbolKind.Property or SymbolKind.Field } member
```

Reply:
That would break for extended property scenario. Will add test to illustrate